pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -456,10 +456,18 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
     int numUnavailableSegments = unavailableSegments.size();
     requestStatistics.setNumUnavailableSegments(numUnavailableSegments);
 
+    List<QueryProcessingException> exceptions = new ArrayList<>();
+    if (numUnavailableSegments > 0) {
+      exceptions.add(new QueryProcessingException(QueryException.BROKER_SEGMENT_UNAVAILABLE_ERROR_CODE,
+          String.format("%d segments %s unavailable", numUnavailableSegments, unavailableSegments)));
+    }
```

Comment:
Here we can store a list of `ProcessingException` which is easier to manage
```suggestion
    List<ProcessingException> exceptions = new ArrayList<>();
    if (numUnavailableSegments > 0) {
      exceptions.add(QueryException.getException(QueryException.BROKER_SEGMENT_UNAVAILABLE_ERROR_CODE,
          String.format("%d segments %s unavailable", numUnavailableSegments, unavailableSegments)));
    }
```

Reply:
The only downside of this is under the hood, `getException` does
```
String errorType = processingException.getMessage();
    ProcessingException copiedProcessingException = processingException.deepCopy();
    copiedProcessingException.setMessage(errorType + ":\n" + errorMessage);
    return copiedProcessingException;
```
which is setting the error message to `"message": "null:\n1 segments [meetupRsvp_REALTIME_1637786040816_0__0__1637786041067] unavailable"`. The `null:\n` looks weird. I could make that code not do that on nulls, but I have no idea what else changes at that point.